source/witch/witch.js

```diff
@@ -10,86 +12,97 @@ const FAILED = "FAILED";
 
 const BUCKET = process.env.BUCKET;
 
-exports.staticHandler = function(event, context) {
-    if (event.RequestType !== "Create" && event.RequestType !== "Update") {
-        return respond(event, context, SUCCESS, {});
-    }
+exports.staticHandler = function (event, context) {
+  if (event.RequestType !== "Create" && event.RequestType !== "Update") {
+    return respond(event, context, SUCCESS, {});
+  }
+
+  Promise.all(
+    walkSync("./").map((file) => {
+      const fileType = mime.lookup(file) || "application/octet-stream";
 
-    Promise.all(walkSync("./").map(file => {
-        var fileType = mime.lookup(file) || "application/octet-stream";
-
-        console.log(`${file} -> ${fileType}`);
-
-        return s3.upload({
-            Body: fs.createReadStream(file),
-            Bucket: BUCKET,
-            ContentType: fileType,
-            Key: file,
-            ACL: "private",
-        }).promise();
-    })).then((msg) => {
-        respond(event, context, SUCCESS, {});
-    }).catch(err => {
-        respond(event, context, FAILED, {Message: err});
+      console.log(`${file} -> ${fileType}`);
+
+      return s3
+        .upload({
+          Body: fs.createReadStream(file),
+          Bucket: BUCKET,
+          ContentType: fileType,
+          Key: file,
+          ACL: "private",
+        })
+        .promise();
+    })
+  )
+    .then((msg) => {
+      respond(event, context, SUCCESS, {});
+    })
+    .catch((err) => {
+      respond(event, context, FAILED, { Message: err });
     });
 };
 
 // List all files in a directory in Node.js recursively in a synchronous fashion
 function walkSync(dir, filelist) {
-	var files = fs.readdirSync(dir);
-	filelist = filelist || [];
-
-	files.forEach(function(file) {
-		if (fs.statSync(path.join(dir, file)).isDirectory()) {
-			filelist = walkSync(path.join(dir, file), filelist);
-		} else {
-			filelist.push(path.join(dir, file));
-		}
-	});
-
-	return filelist;
-};
+  const files = fs.readdirSync(dir);
+  filelist = filelist || [];
```

Comment:
You can get rid of this line by using default paramters, just change the function definition to this:

function walkSync(dir, filelist = []) {

Reply:
Good suggestion!